File: metadata/src/application/dtos/node_schema_dto.rs
use crate::domain::{NodeSchemaIdModel, NodeSchemaModel};
use bric_a_brac_dtos::{NodeSchemaDto, NodeSchemaIdDto};

impl From<NodeSchemaIdModel> for NodeSchemaIdDto {
    fn from(graph_id: NodeSchemaIdModel) -> Self {
        Self::from(*graph_id.as_ref())
    }
}

impl From<NodeSchemaIdDto> for NodeSchemaIdModel {
    fn from(graph_id: NodeSchemaIdDto) -> Self {
        Self::from(*graph_id.as_ref())
    }
}

impl From<NodeSchemaModel> for NodeSchemaDto {
    fn from(model: NodeSchemaModel) -> Self {
        Self {
            node_schema_id: model.node_schema_id.into(),
            graph_id: model.graph_id.into(),
            label: model.label.into(),
            key: model.key.into(),
            color: model.color.into(),
            description: model.description.into(),
            created_at: model.created_at,
            updated_at: model.updated_at,
        }
    }
}

File: web-ui/lib/api/dtos/node-schema-dto.ts
import * as v from "valibot";

export const NodeSchemaDto = v.object({
  node_schema_id: v.string(),
  graph_id: v.string(),
  label: v.string(),
  key: v.string(),
  color: v.string(),
  description: v.string(),
  created_at: v.pipe(v.string(), v.isoTimestamp()),
  updated_at: v.pipe(v.string(), v.isoTimestamp())
});
